src/templates/array3d.h

```diff
@@ -104,6 +104,22 @@ template <class A> class Array3D
 
         return array_[sliceOffsets_[x] + y * nZ_ + z];
     }
+    // Return array range for a given x and y value
+    std::pair<typename std::vector<A>::iterator , typename std::vector<A>::iterator> operator[](std::tuple<int, int> index)
+    {
+        auto [x, y] = index;
+        auto begin = array_.begin() + sliceOffsets_[x] + y * nZ_;
+        auto end = array_.begin() + sliceOffsets_[x] + (y + 1) * nZ_;
+        return {begin, end};
+    }
+    // Return array range for a given x and y value
+    std::pair<typename std::vector<A>::const_iterator , typename std::vector<A>::const_iterator> operator[](std::tuple<int, int> index) const
+    {
+        auto [x, y] = index;
+        auto begin = array_.begin() + sliceOffsets_[x] + y * nZ_;
+        auto end = array_.begin() + sliceOffsets_[x] + (y + 1) * nZ_;
```

Comment:
This is either tiredness or inspiration (or a really bad combination of both), but could one save a bit of math effort and simplify to:
```suggestion
        auto end = begin + nZ_;
```

Reply:
That is truly inspiration.  Having seen it, any other way of writing is now obviously wrong.